src/helics/application_api/ValueFederate.cpp

```diff
@@ -281,9 +281,10 @@ void ValueFederate::registerValueInterfacesToml(const std::string& tomlString)
     }
     bool defaultGlobal = false;
     replaceIfMember(doc, "defaultglobal", defaultGlobal);
-    auto pubs = doc.find("publications");
-    if (pubs != nullptr) {
-        auto& pubArray = pubs->as<toml::Array>();
+
+    if (isMember(doc, "publications")) {
+        auto pubs = toml::find(doc, "publications");
+        auto& pubArray = pubs.as_array();
```

Comment:
Should publications, subscriptions, and inputs (and MessageFederate endpoints) have checks to make sure the toml element is an array?